src/Raven.Server/Documents/Handlers/IndexHandler.cs

```diff
@@ -751,6 +756,117 @@ public Task SuggestIndexMerge()
 
         }
 
+        [RavenAction("/databases/*/indexes/try", "POST", AuthorizationStatus.ValidUser)]
+        public async Task TestJavaScriptIndex()
+        {
+            using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            {
+                var input = await context.ReadForMemoryAsync(RequestBodyStream(), "TestJavaScriptIndex");
+                if (input.TryGet("Definition", out BlittableJsonReaderObject index) == false)
+                    ThrowRequiredPropertyNameInRequest("Definition");
+
+                var indexDefinition = JsonDeserializationServer.IndexDefinition(index);
```

Comment:
Why aren't you just using the root object here?

Reply:
Because nobody responded to my request to define the controller requirement for the studio. Below you want to pass is where should they reside, inside the index definitions?
I expected a million comments on my work I planned ahead.

Reply:
Okay, make sense. We'll need an array of document ids there, for sure.